### PR TITLE
Edit profile data#53

### LINKED ISSUE #53: User Story 19, User Can Edit their Profile Data
```
As a registered user
When I visit my profile page
I see a link to edit my profile data
When I click on the link to edit my profile data
Then my current URI route is "/profile/edit"
I see a form like the registration page
The form contains all of my user information
The password fields are blank and can be left blank
I can change any or all of the information
When I submit the form
Then I am returned to my profile page
And I see a flash message telling me that my data is updated
And I see my updated information
```

### Checklist:

- [ ] base code tests are written
- [ ] base code is written to pass tests
- [ ] edge case ("sad path") tests written (if required)
- [ ] edge case code written (if required)
- [ ] code is reviewed
- [ ] branch is merged
- [ ] all tests still pass

child of #56

depends on #55

### PR DESCRIPTION
**Pull request by asmolentzov**
Completes User Story 19: User Can Edit their Profile Data
Separates registration form into partial view so it can be rendered in both new and edit user views. 
Adds edit path to routes.
Adds edit and update methods to user controller. 
Checks for successful user creation in update method (guards against leaving fields blank).
All tests passing, 100% coverage.
Closes #53

### CODE REVIEW

**Comment by Bradniedt** [comment]
Looks great! This has my approval to merge once another teammate approves. Thanks!!

**Comment by Bradniedt** [comment]
Looks great! Thanks for adding those tests! I approve merging this.

File: app/controllers/users_controller.rb [hunk 1]
```diff
@@ -27,6 +27,12 @@ def create
   def edit
     @user = current_user
   end
+  
+  def update
+    current_user.update(user_params)
+    flash[:success] = 'You have updated your profile'
```
Comment (danbriechle) [inline app/controllers/users_controller.rb]: discussed in person. Looks solid !
